app/src/main/java/com/keylesspalace/tusky/components/report/adapter/StatusesAdapter.kt

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2019 Joel Pyska
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.report.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.AsyncPagedListDiffer
+import androidx.paging.PagedList
+import androidx.recyclerview.widget.AsyncDifferConfig
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListUpdateCallback
+import androidx.recyclerview.widget.RecyclerView
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.components.report.model.StatusViewState
+import com.keylesspalace.tusky.entity.Status
+
+class StatusesAdapter(private val useAbsoluteTime: Boolean,
+                      private val mediaPreviewEnabled: Boolean,
+                      private val statusViewState: StatusViewState,
+                      private val adapterHandler: AdapterHandler)
+    : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```

Comment:
I think this is the perfect usecase for implementing `androidx.recyclerview.widgetListAdapter`. It handles the diffing, `getItemCount()` and `submitList()` for you and has a `getItem` method to get an item for a position so your code gets way shorter.

Reply:
Done